CarpetX/src/schedule.cxx

```diff
@@ -1122,17 +1153,7 @@ int Initialise(tFleshConfig *config) {
     if (CCTK_EQUALS(timestep_choice, "timestep")) {
       cctkGH->cctk_delta_time = timestep;
     } else if (CCTK_EQUALS(timestep_choice, "dtfac")) {
-      CCTK_REAL mindx = 1.0 / 0.0;
-      for (const auto &patchdata : ghext->patchdata) {
-        const amrex::Geometry &geom = patchdata.amrcore->Geom(0);
-        const CCTK_REAL *restrict const dx = geom.CellSize();
-        CCTK_REAL mindx1 = 1.0 / 0.0;
-        for (int d = 0; d < dim; ++d)
-          mindx1 = fmin(mindx1, dx[d]);
-        mindx1 = ldexp(mindx1, -(int(patchdata.leveldata.size()) - 1));
-        mindx = fmin(mindx, mindx1);
-      }
-      cctkGH->cctk_delta_time = dtfac * mindx;
+      cctkGH->cctk_delta_time = dtfac * get_mindx(use_subcycling_wip);
```

Comment:
Whether subcycling is enabled or not is a global setting that doesn't change over time. There is no need to pass this as argument to `get_mindx` which already looks at other settings and other global state.

What does `mindx` mean in this context? Is this still the right name for the function? Is this the minimum spacing of the coarsest level, the current level, the finest level, the finest currently active level, ...?

Maybe it's cleanest to write `use_subcycling ? get_coarse_mindx() : get_finest_mindx()` here.